Common/include/option_structure.inl

```diff
@@ -70,6 +70,51 @@ public:
   }
 };
 
+
+template <class Tenum>
+class COptionEnumClass : public COptionBase {
+
+  map<string, Tenum> m;
+  Tenum & field; // Reference to the fieldname
```

Comment:
Hmmmm not to be picky, but :)
- You can avoid duplicating this class by giving it 2 template parameters, one for the enum type, the other for the type of the field. The first type is always the enum type, the second can be either the conventional unsigned short or the enum type again. Template parameters can have default values, so you even get to do this with minimal changes.
- I never noticed that this class made a copy of the map... please make it a const reference, and pass it by reference also in the constructor.

Reply:
Don't hesitate to be picky 🔍 

Thats an easy and straight forward solution 👍 for the first point. The second was a bit trickier and I had to un`override` the `SetValue` function for now as this line 
```c++
Tenum val = this->m[option_value[0]];
```
is violating the const of m indirectly. So I made the member function `SetValue` const to calm down the compiler but then I had to remove the override to make it compile. I dont fully understand what is going here and there is surely a better way, so my current version is a version-to-be-improved.

Reply:
ok that const ref thing compiled for me but not for github, so I reverted the change as it was not really good anyway...